torchvision/transforms/transforms.py

```diff
@@ -1099,3 +1099,147 @@ def __call__(self, img):
+                flattened array for each channel.
+            noise_range (float): Default is 0. A uniform noise ranged between 
+                (-noise_range, +noise_range) will be added to pixels randomly.
+
+        Returns:
+            histogram transformed tensor: 
+                The output tensor type matches the input, either numpy.ndarray or torch.Tensor.
+        """
+        tensorType       = type(tensor)
+        tensor           = np.asanyarray(tensor)
+        
+        channels = []
+        for c, templateHisto in enumerate(self.template_histograms):
+            channels.append(self.histogram_transform_1D(tensor[c], templateHisto, noise_range = noise_range))
+
+        transformed_tensor = np.asanyarray(channels)
+
+        # Convert to the original type 
+        if tensorType == torch.Tensor:
+            transformed_tensor = torch.tensor(transformed_tensor, dtype=dtype)
+            
+
+        return transformed_tensor
+
+
+    # Core of the computation, to be used by histogram_transform method internally
+    def histogram_transform_1D(self, tensor, template_histogram, noise_range = 0):
+        """
+        Transforms the distribution of the input tensor to match that
+        of the template histogram.
+        
+        Input tensor will be flattened, transformed, and rearranged 
+        to the original shape.
+
+        Mainly intended for call by class functions. 
+        
+        Args:
+            tensor (numpy.ndarray): Image to transform; the histogram is computed 
+                over the flattened array.
+            template_histogram (tubple of (numpy.ndarray, numpy.ndarray)): 
+                The template histogram consisiting of a tuple of (counts, bins). 
+                See (the output of) numpy.histogram. 
+            noise_range (float): Default is 0. A uniform noise ranged between 
+                (-noise_range, +noise_range) will be added to pixels randomly. 
+        Returns:
+            histogram transformed array (numpy.ndarray):
+                The transformed output tensor/image that maches the input
+        """
+        # === Template Histogram ===
+        # t_... stands for template_
+        t_counts, t_bins = template_histogram
+        # t_bin_idx not required
+        
+        # Take the cumsum of the counts and normalize by the number of pixels to
+        # Get the empirical cumulative distribution functions
+        # (maps value --> quantile)
+        t_quantiles      = np.cumsum(t_counts).astype(np.float32)
+        t_quantiles     /= t_quantiles[-1]
+        
+        # === Input Tensor ===
+        # Convert to flattened numpy array
+        tensor           = np.asanyarray(tensor)
+        originalShape    = tensor.shape
+        tensor           = tensor.ravel()
+        
+        # Get counts, bins, and corresponding bin indices for each tensor value
+        counts, bins     = np.histogram(tensor, bins = t_bins)
```

Comment:
Can you change this to torch equivalent [torch.histc()](https://pytorch.org/docs/master/torch.html?highlight=hist#torch.histc)?

Reply:
Unfortunately not; mainly because np.histogram accepts variable bin interval while torch.histc can only handle equal bin interval. Also, that would require conversion between np.array and torch.tensor multiple times.